packages/mui-base/src/Select/Trigger/SelectTrigger.tsx

```diff
@@ -0,0 +1,129 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { useSelectTrigger } from './useSelectTrigger';
+import { useSelectRootContext } from '../Root/SelectRootContext';
+import { commonStyleHooks } from '../utils/commonStyleHooks';
+import { useComponentRenderer } from '../../utils/useComponentRenderer';
+import { BaseUIComponentProps } from '../../utils/types';
+import { useFieldRootContext } from '../../Field/Root/FieldRootContext';
+import { useEnhancedEffect } from '../../utils/useEnhancedEffect';
+
+const SelectTrigger = React.forwardRef(function SelectTrigger(
+  props: SelectTrigger.Props,
+  forwardedRef: React.ForwardedRef<HTMLElement>,
+) {
+  const {
+    render,
+    className,
+    id: idProp,
+    disabled: disabledProp = false,
+    label,
+    ...otherProps
+  } = props;
+
+  const {
+    ownerState: fieldOwnerState,
+    disabled: fieldDisabled,
+    setDisabled,
+  } = useFieldRootContext();
+
+  const {
+    getTriggerProps: getRootTriggerProps,
+    disabled: selectDisabled,
+    open,
+  } = useSelectRootContext();
+
+  const disabled = fieldDisabled || selectDisabled || disabledProp;
+
+  useEnhancedEffect(() => {
+    setDisabled(disabled);
+  }, [disabled, setDisabled]);
+
+  const { getTriggerProps } = useSelectTrigger({
+    disabled,
+    rootRef: forwardedRef,
+  });
+
+  const ownerState: SelectTrigger.OwnerState = React.useMemo(
+    () => ({
+      ...fieldOwnerState,
+      open,
+    }),
+    [fieldOwnerState, open],
+  );
+
+  const { renderElement } = useComponentRenderer({
+    render: render ?? 'div',
+    className,
+    ownerState,
+    propGetter: (externalProps) => getTriggerProps(getRootTriggerProps(externalProps)),
+    customStyleHookMapping: commonStyleHooks,
+    extraProps: otherProps,
```

Comment:
It would be great if the Trigger had a CSS variable with the width of the popup, so it can match it (as in native OS controls)

Reply:
Do you mean the inverse? `--anchor-width` for popup is available. The trigger can't know the width of the popup when it's not mounted

Reply:
No, I didn't mean the inverse. Native OS selects adapt their width to the longest option. Perhaps this could be an option when keepMounted=true.

Reply:
That's true, but I'm hesitant to add something that doesn't work by default. Furthermore,  it's likely very performance intensive with longer lists to calculate as you'll need to measure every option with `getBoundingClientRect` to calculate the largest one. It also doesn't work during SSR, so will cause layout shift. It's likely something that should be hardcoded by the user if necessary.